includes/admin/class-amp-cache-utilities.php

```diff
@@ -0,0 +1,251 @@
+<?php
+abstract class AMP_Cache_Utilities {
+
+	static $amp_valid_content_types = array( 'c', 'i', 'r' );
+	static $amp_cache_url_base = 'https://cdn.ampproject.org';
+	static $amp_cache_update_url_base = 'https://cdn.ampproject.org/update-ping';
+
+	/**
+	 * Hooks the 'post_updated' and 'before_delete_post' actions to determine if we
+	 * need to update the AMP Cache when a post is updated or about to be deleted.
+	 */
+	public static function amp_add_cache_update_actions() {
+		// Hooking this to the post_updated action so this will fire any time a post
+		// is updated in any way (including status transitions).
+		add_action( 'post_updated',  array( 'AMP_Cache_Utilities', 'post_updated' ), 10, 3 );
+		
+		// Hooking the call to update the cache *before* the post is updated in case we need
+		// to access any metadata for future functionality.
+		add_action( 'before_delete_post',  array( 'AMP_Cache_Utilities', 'do_amp_update_ping' ) );
```

Comment:
Now using `post_supports_amp` to make sure the post supports AMP. Also returning early if `post_status` is not `publish` when the `before_delete_post` action is called.